src/parquet/encodings/dictionary-encoding.h

```diff
@@ -418,6 +432,23 @@ inline void DictEncoder<FixedLenByteArray>::WriteDict(uint8_t* buffer) {
   }
 }
 
+template<typename T>
+inline void DictEncoder<T>::FreeMemory() {}
+
+template<>
+inline void DictEncoder<ByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
+
+template<>
+inline void DictEncoder<FixedLenByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
```

Comment:
I am still pretty uncomfortable with this. I don't think we should ever use other memory allocators than the chunked one in dictionary encoding -- if DictEncoder is initialized with a normal memory pool, it can initialize the ChunkedMemoryPool (which has the passed MemoryPool as a member variable, rather than using inheritance, to chunk memory allocations, then freeing them all once complete).